<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsAttributeTypeDialog</class>
 <widget class="QDialog" name="QgsAttributeTypeDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>751</width>
    <height>487</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit Widget Properties</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="5" column="1">
    <widget class="QStackedWidget" name="stackedWidget"/>
   </item>
   <item row="4" column="1">
    <widget class="QgsCollapsibleGroupBox" name="groupBox">
     <property name="title">
      <string>Constraints</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QCheckBox" name="notNullCheckBox">
        <property name="text">
         <string>Not null</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Constraint</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QgsFieldExpressionWidget" name="constraintExpressionWidget" native="true">
          <property name="focusPolicy">
           <enum>Qt::StrongFocus</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Constraint description</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="leConstraintExpressionDescription"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QCheckBox" name="isFieldEditableCheckBox">
     <property name="text">
      <string>Editable</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Defaults</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2" columnstretch="0,1">
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Default value</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Preview</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QgsExpressionLineEdit" name="mExpressionWidget" native="true">
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="mDefaultPreviewLabel">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QCheckBox" name="labelOnTopCheckBox">
     <property name="text">
      <string>Label on top</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="0" column="0" rowspan="7">
    <widget class="QListWidget" name="selectionListWidget"/>
   </item>
   <item row="6" column="1">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsExpressionLineEdit</class>
   <extends>QWidget</extends>
   <header>qgsexpressionlineedit.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsFieldExpressionWidget</class>
   <extends>QWidget</extends>
   <header>qgsfieldexpressionwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>selectionListWidget</tabstop>
  <tabstop>isFieldEditableCheckBox</tabstop>
  <tabstop>labelOnTopCheckBox</tabstop>
  <tabstop>mExpressionWidget</tabstop>
  <tabstop>notNullCheckBox</tabstop>
  <tabstop>constraintExpressionWidget</tabstop>
  <tabstop>leConstraintExpressionDescription</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>QgsAttributeTypeDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>520</x>
     <y>435</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>QgsAttributeTypeDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>588</x>
     <y>435</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
